packages/core/core/src/requests/AssetRequest.js

```diff
@@ -0,0 +1,139 @@
+// @flow strict-local
+import type WorkerFarm from '@parcel/workers';
+import type {StaticRunOpts} from '../RequestTracker';
+import type {
+  Asset,
+  AssetRequestInput,
+  AssetRequestResult,
+  Config,
+  ConfigRequestDesc,
+  TransformationOpts,
+} from '../types';
+
+import {md5FromObject} from '@parcel/utils';
+import invariant from 'assert';
+import nullthrows from 'nullthrows';
+
+type RunInput = {|
+  input: AssetRequestInput,
+  ...StaticRunOpts,
+|};
+
+export type AssetRequest = {|
+  id: string,
+  +type: 'asset_request',
+  run: RunInput => Promise<AssetRequestResult>,
+  input: AssetRequestInput,
+|};
+
+function generateRequestId(type, obj) {
+  return `${type}:${md5FromObject(obj)}`;
+}
+
+export default function createAssetRequest(input: AssetRequestInput) {
+  return {
+    type: 'asset_request',
+    id: getId(input),
+    run,
+    input,
+  };
+}
+
+let handle;
+function getRunTransform(
+  farm: WorkerFarm,
+): TransformationOpts => Promise<{|
+  assets: Array<Asset>,
+  configRequests: Array<{|request: ConfigRequestDesc, result: Config|}>,
+|}> {
+  if (handle != null) {
+    return handle;
+  }
+
+  handle = farm.createHandle('runTransform');
+  return handle;
+  // ? Could this singleton cause problems
```

Comment:
Yeah, I'll fix this.

- [ ] Remove singleton workerfarm handle

Reply:
I thought about creating a cached handle map in either the WorkerFarm or RequestTracker, but I decided to try out creating a new handle function in each request and I didn't notice any regressions in build times or memory usage. I've pushed that since it's the simplest solution, but let me know if you foresee issues with that approach.